src/solver/fixed_shaped_solver/fixed_shaped_solver.jl

```diff
@@ -0,0 +1,342 @@
+"""
+Representation of a branching constraint in the search tree.
```

Comment:
This description could be a bit more elaborate. Or you include a longer description elsewhere where you give an intuition where and how to use these sub-structs.